`@MockkBean` in the constructor doesn't play well with `verifyAll`

The following test fails on `java.lang.AssertionError: Verification failed: some calls were not matched: [TestService(com.example.playground.TestService#0 bean#1).equals(TestService(com.example.playground.TestService#0 bean#1))]
`:
```
@WebMvcTest
class PlaygroundApplicationTests @Autowired constructor(@MockkBean val testService: TestService) {

    @Test
    fun testService() {
        every { testService.one() } returns 2
        Assertions.assertEquals(2, testService.one())
        verifyAll { testService.one() }
    }
}

@Service
class TestService {
    fun one() = 1
}
```

If the `@MockkBean` declaration is moved to the class body the test passes as expected:
```
@WebMvcTest
class PlaygroundApplicationTests {

    @MockkBean
    lateinit var testService: TestService

    @Test
    fun testService() {
        every { testService.one() } returns 2
        Assertions.assertEquals(2, testService.one())
        verifyAll { testService.one() }
    }
}
```

--- a/src/main/kotlin/com/ninjasquad/springmockk/MockkPostProcessor.kt
+++ b/src/main/kotlin/com/ninjasquad/springmockk/MockkPostProcessor.kt
@@ -361,7 +361,7 @@ class MockkPostProcessor(private val definitions: Set<Definition>) : Instantiati
             field.isAccessible = true
             val existingValue = ReflectionUtils.getField(field, target);
             val bean = this.beanFactory.getBean(beanName, field.type)
-            if (existingValue == bean) {
+            if (existingValue === bean) {
                 return;
             }
             check(existingValue == null) {

--- a/src/test/kotlin/com/ninjasquad/springmockk/VerifyAllTests.kt
+++ b/src/test/kotlin/com/ninjasquad/springmockk/VerifyAllTests.kt
@@ -1,0 +1,31 @@
+package com.ninjasquad.springmockk
+
+import io.mockk.every
+import io.mockk.verifyAll
+import org.assertj.core.api.Assertions
+import org.assertj.core.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import org.springframework.test.context.junit.jupiter.SpringExtension
+
+/**
+ * Tests for issue https://github.com/Ninja-Squad/springmockk/issues/90
+ * @author JB Nizet
+ */
+@ExtendWith(SpringExtension::class)
+class VerifyAllTests @Autowired constructor(@MockkBean val testService: VerifyAllTestService) {
+
+    @Test
+    fun testService() {
+        every { testService.one() } returns 2
+        assertThat(testService.one()).isEqualTo(2)
+        verifyAll { testService.one() }
+    }
+}
+
+@Component
+class VerifyAllTestService {
+    fun one() = 1
+}

